src/sagas/index.js

```diff
@@ -343,19 +390,39 @@ export function* startProgress ({ vmId, poolId, name }) {
 }
 
 export function* stopProgress ({ vmId, poolId, name, result }) {
-  if (result && result.status === 'complete') {
-    vmId = vmId || result.vm.id
-    // do not call 'end of in progress' if successful,
-    // since UI will be updated by refresh
-    yield delay(5 * 1000)
-    yield getSingleInstance({ vmId, poolId })
-
-    yield delay(30 * 1000)
-    yield getSingleInstance({ vmId, poolId })
+  if (result?.status === 'complete' && result?.job?.id) {
+    // If the result is complete, refresh the entity to grab the current status
+    if (vmId) {
+      yield put(getSingleVm({ vmId, shallowFetch: true }))
+    } else {
+      yield put(getSinglePool({ vmId }))
+    }
+
+    // If the result has an async job associated with it, wait
+    // for the job to finish then refresh the VM or Pool again
+    const jobId = result.job.id
+
+    for (const delayMs of delayInMsSteps()) {
```

Comment:
> Previously there were 2 requests ( after 5 sec and 30sec) with total wait time 35sec. After the change there will be up to 20 requests with delay from 2sec to 9 sec with total wait time up to ~130sec.
> 
>     1. the initial delay seems too small

Quite often the request for the first job will return it as complete.

> 
>     2. in first 35sec we are doing 10 requests - it seems too much if previously 2 were OK

Those 2 were vestigial and didn't do anything really useful for us.  It was far more common to have to wait for the normal background refresh timer to see any VM state change.  This loop actually puts the check for status change behavior of VM actions in line with the VM create and Disk create/change status checking loops.

In my own testing, the updates felt a lot smoother with this pattern in place.

Reply:
(moving these sagas to `vmChanges.js` to be with the sagas that use them)

Reply:
> Quite often the request for the first job will return it as complete.
>
vs
> It was far more common to have to wait for the normal background refresh timer to see any VM state change. 
>

I assume that there is some truth in both of the above statements :)  Sometimes the job will run fast (almost instantly) and sometimes it may take minutes. IMHO we should tweak the number of requests and timing to take that into account. Firing too many queries in the early phase (every ~2sec) seems to heavy.